### ISSUE TITLE
[BetterBibTeX Startup error] better bibtex: waiting for zotero 

### ISSUE THREAD

**Issue by Clue-ess-coder** [retorquere/zotero-better-bibtex#2995]
### Debug log ID

DP2I6EVZ-fio-Plta6dDidtLe

### What happened?

I used Zotero 6 with better bibtex without issues. Upgraded to Zotero 7 and had to reinstall Better BibTeX, for some reason it wasn't exporting my library and I just REALLY hated Zotero 7's UI, so I switched back to Zotero 6. (exporting the library didn't work in Zotero 7, but I think the plugin did startup correctly). And now, after reinstalling better bibtex in Zotero 6, there seems to be a startup error. 

I attached the debug log ID. 

From the console this seems to be the issue:

> {better-bibtex-startup} BetterBibTeX startup error: A parameter or an operation is not supported by the underlying object transaction@chrome://zotero-better-bibtex/content/better-bibtex.js:15824:34 open@chrome://zotero-better-bibtex/content/better-bibtex.js:16156:24 

There are no sync conflicts between my local library and zotero servers.

## I tried

- I tried disabling and removing the add-on (restarting zotero in both cases), didn't work in both cases.
- I also toggled `automaticScraperUpdates` from `true` to `false` and vice-versa (restarting zotero in both cases), still didn't work.

**Comment by retorquere** [comment]
Please try build 7051

> I also toggled automaticScraperUpdates from true to false and vice-versa (restarting zotero in both cases), still didn't work.

I don't know what that is intended to do.

**Comment by Clue-ess-coder** [comment]
I think I messed up the database or something, since I installed Zotero 6 on top of Zotero 7 (I didn't uninstall Zotero 6 first). I fixed that, cleaned up everything, and did a clean install of Zotero 7. Now, we're good. 👍

**Comment by retorquere** [comment]
> I think I messed up the database or something, since I installed Zotero 6 on top of Zotero 7 (I didn't uninstall Zotero 6 first). I fixed that, cleaned up everything, and did a clean install of Zotero 7. Now, we're good.

You didn't mess anything up, it is quite possible that the caching system I use isn't downwards compatible. Build 7051 was supposed to address that.

**Comment by Clue-ess-coder** [comment]
Would downwards compatibility be supported in future releases? (As there's a very high I'll switch back to Zotero 6, LOL—Zotero 7 doesn't quite do the job for me.)

**Comment by retorquere** [comment]
That's what build 7051 addressed.

**Comment by Clue-ess-coder** [comment]
Okay so, switched back to Zotero 6, and yes better bibtex starts up as expected. But it seems there's a new problem. Automatic exports don't seem to be working. I tried: exporting with notes, without notes, different Better-<formats> and from Better BibTeX preferences with `[Export now]`, I noticed the `updated`  field didn't change at all: It's stuck at the export's created time.

New items are updated in the export (although the `updated` field still doesn't reflect that); notes aren't reflected still.

**Comment by retorquere** [comment]
debug log id

**Comment by Clue-ess-coder** [comment]
Here you go: P3TYAS2D-fio-dYXfOtbYPziB

**Comment by retorquere** [comment]
Did you mean to close the issue?

**Comment by Clue-ess-coder** [comment]
No, sorry. 😅

**Comment by retorquere** [comment]
A new build will drop in a few minutes, please reproduce the problem and send a new log (regular log preferred over fio, if the regular log works).

**Comment by Clue-ess-coder** [comment]
Very well then.

**Comment by Clue-ess-coder** [comment]
No changes still.

[Debug Output.txt](https://github.com/user-attachments/files/17094118/Debug.Output.txt)

**Comment by retorquere** [comment]
Doesn't the regular BBT debug log work?

**Comment by Clue-ess-coder** [comment]
Ah you meant THAT log, here is the ID: DD5K5DXQ-euc/6.7.236.7054-6

**Comment by retorquere** [comment]
Can you recreate the problem, send a new debug log, and check the "include cache" checkbox?

**Comment by Clue-ess-coder** [comment]
Here you go: VZATSRSI-euc/6.7.236.7080-6

**Comment by Clue-ess-coder** [comment]
After installing build 7080: VZATSRSI-euc/6.7.236.7080-6

After test: 89DFFR97-euc/6.7.236.7080-6

Export seems to work when I explicitly click `[Export now]` in preferences. The `updated` field is stuck at the created time still. And there seems to be an issue with syncing with Zotero servers now.

**Comment by retorquere** [comment]
Are you sure the latter is BBT related? I see errors relating to sync in your log, but that's about accessing https://api.zotero.org/keys/current, and if I access that outside Zotero, I get timeout errors too. It seems like something is amiss with their servers.

**Comment by Clue-ess-coder** [comment]
Apparently, it wasn't. I think my internet had issues or something; Zotero sync works now. 

While It doesn't compare to the previous working of better BibTeX (before the Zotero 7 shenanigans), but it works. Thank you! 🎉

**Comment by retorquere** [comment]
I didn't really need the work it involved but 7 I a lot faster all around, also BBT is faster, and it solves memory problems on Windows. The main reason BBT was hard to port is that it does so many things that Zotero wasn't prepared for, so I have deeply zotero-version-dependent code.

**Comment by retorquere** [comment]
Can you select a few of the items that have notes that are not being exported, right-click, and then send a debug log from the popup menu? The log ID will have `-refs` in it.

**Comment by Clue-ess-coder** [comment]
Sorry for responding late.

Let me rephrase the issue as of right now.

The only time export seems to work is when I use the `Add Note from Annotations` option. My workflow right now involves me extracting *some* of those highlights, which means I have to manually create an item note, then add the notes/highlights I want to export with the `Add to Note` option.

`[Export now]` from BBT preferences seems to be the only way for me to export *these* notes.

I hope that makes sense.
